### ISSUE TITLE
Recordings not Converting 

### ISSUE THREAD

**Issue by alihastie** [ndunand/moodle-filter_html5avtomp4#6]
Hi

We have installed the plugin on Moodle 3.8 version and the current and new recordings through atto_recordrtc have not converted to MP4 sources for HTML5 <audio> and <video> elements. Tested on both iPhone and iPad with the ' No compatible source was found for this media' message.

We have setup the path to ffmpeg correctly, and kept with the convert audio and video settings as is.

We also have scheduling set to run every minute within Moodle for the plugin.

**Comment by ndunand** [comment]
Hlelo,

First, please make sure the plugin is actually enabled.

Then, you go into Moodle administration > Server > Tasks > Scheduled tasks, look for "filter_html5avtomp4", there should be a task definition here with a "Run now" (tiny) link. Try running it manually and please post the results here.

**Comment by alihastie** [comment]
Thanks for the prompt reply.

The plugin has been enabled on the Filters settings, and within the course areas.

We do not see a "Run now" link within schedules tasks. 

![html5filter](https://user-images.githubusercontent.com/9861902/102196203-78b50a80-3eb7-11eb-9d62-1a764cdadd33.PNG)

We also have the plugin on another Moodle 3.8 installation, with the exact same settings and we can see the audio/video players on an iPhone, but they don't play:

![iphone_html5](https://user-images.githubusercontent.com/9861902/102199069-13631880-3ebb-11eb-8c6e-939ece3dc434.jpg)

Thanks

**Comment by ndunand** [comment]
Ah, alright apparently the "Run Now" option has to be set up: https://docs.moodle.org/310/en/Scheduled_tasks#Running_individual_tasks .

iPhone player visible but don't play is probably because the audio or video has not been converted to an iOS-compatible format.

**Comment by alihastie** [comment]
Thanks for the heads up on Run now. 

Ran schedule task on original Moodle instance and received this:

Execute scheduled task: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)
found 0 jobs
... used 1 dbqueries
... used 0.0072319507598877 seconds
Scheduled task complete: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)

Still does not convert through using Safari on iPhone.

**Comment by ndunand** [comment]
Thanks,

Could you then:

1. deactivate the task
2. add a video using atto_recordrtc from Chrome
3. run the task manually

This was ensures there actually is something to convert within the task.

**Comment by alihastie** [comment]
Followed the steps above, though 'disabled' task and to enable to run the task manually:

Execute scheduled task: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)
... started 11:16:10. Current memory use 41.8MB.
found 0 jobs
... used 1 dbqueries
... used 0.00059700012207031 seconds
Scheduled task complete: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)

**Comment by ndunand** [comment]
Strange ... how is the plugin configured exactly ?

**Comment by alihastie** [comment]
![htmlsconfig](https://user-images.githubusercontent.com/9861902/102210975-cd15b580-3eca-11eb-9419-4b80b84e622a.PNG)

**Comment by ndunand** [comment]
Do you have any rows in the `mdl_filter_html5avtomp4_jobs` DB table, and does this number increase after you add a video?

**Comment by alihastie** [comment]
Our Moodle installation is externally hosted. I could raise a support ticket to find out more information?

**Comment by ndunand** [comment]
Yes that would help. I just tested again on both Moodle 3.7 and 3.9 with no issues as I have no 3.8 at hand, just to make sure. It should work on 3.8 as well.

**Comment by alihastie** [comment]
Our hosting company has checked the rows in the database and stated there are no entries. His comment: 'I've tested the record and DB insert, but there does not seem to be any inserts showing within the database table when I create a video.'

Cheers

**Comment by barry-matthias** [comment]
we have also found this issue as a bug
the db table mdl_filter_html5avtomp4_jobs has entries present all with the status of 2 (no idea what that means) at the time
we checked the filters local lib and found status 2 means its completed the conversion so we are now checking if this has actually happened from the user point of view

**Comment by Danie** [comment]
I've testet the plugin on 3.8, 3.9 and 3.10 - it works on all installations without any trouble if the following is considered:

The filter **HTML5 audio/video to MP4 filter** should be placed at least before filters **Convert URLs into links and images** and **Multimedia plugins** (see attached file). If the **HTML5 audio/video to MP4 filter** is behind these two filters, then both the on-the-fly audio conversion and the video conversion via cron job will not work.


![filter_order](https://user-images.githubusercontent.com/917484/105409729-2c38b980-5c31-11eb-8c25-5f391c21acd3.png)

**Comment by ndunand** [comment]
Well spotted @Danie !

Indeed the filter order has an impact, as e.g. this filter expects the test (html) to be formed a certain way, so of course if the text has been changed by another filter upstream this filter can then not do its job as expected.

@alihastie could this resolve your issue?

**Comment by alihastie** [comment]
We have adjusted the Filter options as the same as Danie has in the screenshot above.

Though, the players (audio and video) do display on an iPhone for example which is progress, they will not play. And, as you can see the timer for the video below, it displays 00.00, audio displays 0.00
![IMG_1805](https://user-images.githubusercontent.com/9861902/105494519-d60c5a80-5cb2-11eb-89ca-d40646bbca3a.PNG)

**Comment by Danie** [comment]
Do you have run the moodle-cronjob (/usr/bin/php /path/to/moodle/admin/cli/cron.php)?

**Comment by alihastie** [comment]
Our cronjob runs every 5 minutes, we are unable to run the cron through the Browser.

Just ran HTML5 scheduled task, which displayed this:

Process reencoding non-MP4 files
Execute scheduled task: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)
found 0 jobs
... used 1 dbqueries
... used 0.012264966964722 seconds
Scheduled task complete: Process reencoding non-MP4 files (filter_html5avtomp4\task\processjobs_task)

**Comment by ndunand** [comment]
@alihastie just to rule this possibility out, could you try to just disable all other filters on this specifig Moodle page?

**Comment by Danie** [comment]
@alihastie give the filter a place on top of the filter list.

**Comment by alihastie** [comment]
Disabled all filters,and placed the filter at the top of the list which still displays the players, and still set to 0.00 and won'y play.

**Comment by ndunand** [comment]
Strange ... is there any way you could give me access to a page showing this issue? Failing that, at least post an image of the browser dev tools showing the exact contents of the `<video>` tag?

**Comment by alihastie** [comment]
![videotag](https://user-images.githubusercontent.com/9861902/105849689-a02fe480-5fd8-11eb-95a7-fb25c57c0844.PNG)

**Comment by ndunand** [comment]
@alihastie sorry for letting this wait for so long. By any chance, does the updated version of the plugin solve this issue ? If not, could you post the contents of the `<video>` tag (as you did in the screenshot above, but with the video tag unfolded)?
